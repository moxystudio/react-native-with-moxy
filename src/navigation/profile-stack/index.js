import React from 'react';
import { createStackNavigator } from '@react-navigation/stack';
import ProfileHeader from './header';
import ProfileScreen1 from '../../screens/profile/1/ProfileScreen1';
import ProfileScreen2 from '../../screens/profile/2/ProfileScreen2';

const Stack = createStackNavigator();

const ProfileStack = () => (
    <Stack.Navigator screenOptions={ {
        header: ProfileHeader,
    } }>
        <Stack.Screen name="Profile1" component={ ProfileScreen1 } />
        <Stack.Screen name="Profile2" component={ ProfileScreen2 } />
    </Stack.Navigator>
);

export default ProfileStack;
