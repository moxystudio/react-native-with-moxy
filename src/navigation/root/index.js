import * as React from 'react';

export const navigationRef = React.createRef();

export function navigate(name, params) {
    /* eslint-disable-next-line babel/no-unused-expressions */
    navigationRef.current?.navigate(name, params);
}
