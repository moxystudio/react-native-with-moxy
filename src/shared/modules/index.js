/* istanbul ignore file */

export { default as Button } from './react-native-button';
export { default as SplashScreen } from './react-native-splash-screen';
export { default as SafeAreaHeader } from './react-native-safe-area-header';
export * from './react-theme';
export * from './react-native-locale';
export * from './react-native-app-config';
