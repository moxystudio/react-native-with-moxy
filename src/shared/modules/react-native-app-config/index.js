import AppConfigContext from './context';

const AppConfigConsumer = AppConfigContext.Consumer;

export { default as AppConfigProvider } from './AppConfigProvider';
export { AppConfigConsumer };
export { default as useAppConfig } from './use-app-config';
export { default as withAppConfig } from './with-app-config';
