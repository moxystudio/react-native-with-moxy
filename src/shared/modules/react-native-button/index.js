/* istanbul ignore file */

export { default } from './Button';
