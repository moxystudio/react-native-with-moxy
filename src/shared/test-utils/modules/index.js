export { AppTree } from './react-app-tree';
