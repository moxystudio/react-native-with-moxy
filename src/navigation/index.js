import React from 'react';
import * as rootNavigation from './root';
import { createStackNavigator } from '@react-navigation/stack';

import ProfileStack from './profile-stack';
import HomeScreen from '../screens/home/HomeScreen';

const Stack = createStackNavigator();

const AppStack = () => (
    <Stack.Navigator headerMode="screen">
        <Stack.Screen name="Home" component={ HomeScreen } />
        <Stack.Screen name="Profile" component={ ProfileStack } options={ { header: () => null } } />
    </Stack.Navigator>
);

export default AppStack;

export {
    rootNavigation,
};
