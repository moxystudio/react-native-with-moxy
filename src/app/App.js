import 'react-native-gesture-handler';
import React from 'react';
import { ThemeProvider } from '../shared/components';
import AppStack, { rootNavigation } from '../navigation';
import { NavigationContainer } from '@react-navigation/native';

const App = () => (
    <ThemeProvider>
        <NavigationContainer ref={ rootNavigation.navigationRef }>
            <AppStack />
        </NavigationContainer>
    </ThemeProvider>
);

export default App;
