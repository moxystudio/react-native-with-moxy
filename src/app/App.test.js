import React from 'react';
import { SafeAreaProvider } from 'react-native-safe-area-context';
import { mount } from 'enzyme';
import localeConfig from '../../intl';
import { rootNavigation } from '../navigation';
import { LocaleProvider, ThemeProvider } from '../shared/modules';
import App from './App';

it('should create navigation ref', () => {
    mount(<App />);

    expect(rootNavigation.navigationRef.current).toEqual(expect.objectContaining({
        navigate: expect.any(Function),
    }));
});

it('should render locale provider', () => {
    const tree = mount(<App />);
    const provider = tree.find(LocaleProvider);

    expect(provider.exists()).toBe(true);
    expect(provider.props()).toEqual(expect.objectContaining(localeConfig));
});

it('should render theme provider', () => {
    const tree = mount(<App />);

    expect(tree.find(ThemeProvider).exists()).toBe(true);
});

it('should render safe area provider', () => {
    const tree = mount(<App />);

    expect(tree.find(SafeAreaProvider).exists()).toBe(true);
});

