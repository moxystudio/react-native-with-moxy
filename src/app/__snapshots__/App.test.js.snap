// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<SafeAreaProvider>
  <ThemeContextProvider>
    <ForwardRef(NavigationContainer)>
      <AppStack />
    </ForwardRef(NavigationContainer)>
  </ThemeContextProvider>
</SafeAreaProvider>
`;
